packages/firestore/src/core/sync_engine.ts

```diff
@@ -1025,7 +1026,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
       );
       const query = await this.localStore.getQueryForTarget(targetId);
       assert(!!query, `Query data for active target ${targetId} not found`);
```

Comment:
I think we could add an `assertNotNull` type guard that that would make the `!` below unnecessary.

Something like:
```ts
assertNotNull<T>(value : T | null, explanation: string): arg is T {
  assert(!!value, explanation);
  return true;
}
```

Reply:
Since these type guards only work in if-statements, this would change the calling site to:

```
  if (assertNotNull(query, `Query data for active target ${targetId} not found`)) {
        const queryData = await this.localStore.allocateQuery(query);
        await this.initializeViewAndComputeSnapshot(
            queryData,
            /*current=*/ false
        );
        this.remoteStore.listen(queryData);
      };
```

If we didn't have to put the `explanation`, then I would be all in favor. But with the explanation this looks a tad strange. If we do drop the explanation, we will never be able to trace these asserts back in minified code.